datumaro/util/scope.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from contextlib import ExitStack, contextmanager
+from functools import partial, wraps
+from typing import Any, Callable, ContextManager, Dict, Optional, Tuple, TypeVar
+import threading
+
+from attr import attrs
+
+from datumaro.util import optional_arg_decorator
+
+T = TypeVar('T')
+
+class Scope:
+    """
+    A context manager that allows to register error and exit callbacks.
+    """
+
+    _thread_locals = threading.local()
+
+    @attrs(auto_attribs=True)
+    class ExitHandler:
+        callback: Callable[[], Any]
+        ignore_errors: bool = True
+
+        def __exit__(self, exc_type=None, exc_value=None, exc_traceback=None):
```

Comment:
```suggestion
        def __exit__(self, exc_type, exc_value, exc_traceback):
```

`__exit__` is always called with three arguments, so the default values are unnecessary.